### PR TITLE
feat: add functionality to verify token from ALB and add on_user_loaded hook

### PR DESCRIPTION
**Pull request by puehringer**
### Developer Checklist (Definition of Done)

**Issue**

- [x] All acceptance criteria from the issue are met
- [x] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [ ] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted
- [x] Unit tests are written (frontend/backend if applicable)
- [x] Integration tests are written (if applicable)

**PR**

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., *bug*, *feature*) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes

- Verifies the signature of an ALB received token with the official KIDs. Previously, it was never verified (which is still fine as the ALB sits before), but if we ever allow passing tokens directly to the backend, we want to make sure to avoid this attack vector. 
  - Requests the public keys for a region/key id, caches it, and uses it to verify the signature.
  - This is *NOT* a breaking change, it should not at all break things if valid tokens are passed.
  - In a future PR, the same should be done for the OAuth2SecurityStore, which uses OAuth2 proxy tokens. These are currently also not verified. 
- Add a `on_user_loaded` hook to react to user logins, which can be used to fetch metadata, log information, update local databases, ...

### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

### CODE REVIEW

File: visyn_core/security/store/alb_security_store.py [hunk 1]
```diff
@@ -9,25 +14,44 @@
 _log = logging.getLogger(__name__)
 
 
+@lru_cache
+def _get_public_key(region: str, kid: str) -> str:
+    return httpx.get(f"https://public-keys.auth.elb.{region}.amazonaws.com/{kid}").text
```
Comment (thinkh) [inline visyn_core/security/store/alb_security_store.py:19]: Initially, it was unclear to me what `kid` is. I would expand it:

```suggestion
def _get_public_key(region: str, key_id: str) -> str:
    return httpx.get(f"https://public-keys.auth.elb.{region}.amazonaws.com/{key_id}").text
```

**Review by thinkh — DISMISSED** [review]
Looks good to me. A minor suggestions from my side below.

File: visyn_core/security/store/alb_security_store.py (hunk 1 above)
Comment (puehringer) [reply visyn_core/security/store/alb_security_store.py]: Done, renamed it and added it to the comment.

**Review by thinkh — DISMISSED** [review]
Thanks.

**Pull request merged**